src/Tasks/Microsoft.NET.Build.Tasks/ResolveAppHosts.cs

```diff
@@ -188,6 +188,13 @@ protected override void ExecuteCore()
             {
                 foreach (var otherRuntimeIdentifier in OtherRuntimeIdentifiers)
                 {
+                    // The 'any' RID represents a platform-agnostic platform. As such, it has no
```

Comment:
Should we consider the case where PackAsToolShimRuntimeIdentifiers contains 'any'? In this case, GetEmbeddedAppHostPaths may try to create an appHostDestinationFilePath for the 'any' rid.

Reply:
IMO no - it is an error for `PackAsToolShimRuntimeIdentifiers` to ever contain `any` as there is no way for us to logically create an `any` shim apphost. I could see that being a useful validation check in the PackTool targets, though - what do you think about that?